br/pkg/checksum/executor_test.go

```diff
@@ -15,56 +15,167 @@
 package checksum
 
 import (
+	"bytes"
+	"context"
+	"fmt"
+	"hash/crc64"
+	"sort"
 	"testing"
 
+	"github.com/pingcap/kvproto/pkg/kvrpcpb"
 	"github.com/stretchr/testify/require"
+	"github.com/tikv/client-go/v2/rawkv"
+	"github.com/tikv/migration/br/pkg/backup"
+	"github.com/tikv/migration/br/pkg/utils"
 )
 
-/*
-type mockPdCLient struct {
-	pd.Client
+type TestChecksumClient struct {
+	store map[string]string
 }
 
-func getStubChecksum() Checksum {
-	return Checksum{
-		Crc64Xor:   1234,
-		TotalKvs:   10,
-		TotalBytes: 100,
+func (c *TestChecksumClient) PutBatch(ctx context.Context, keys, values []string) {
+	for i, key := range keys {
+		c.store[key] = values[i]
 	}
 }
 
-func (m *mockPdCLient) RawChecksum(ctx context.Context, in *kvrpcpb.RawChecksumRequest, opts ...grpc.CallOption) (*kvrpcpb.RawChecksumResponse, error) {
-	stubChecksum := getStubChecksum()
-	response := kvrpcpb.RawChecksumResponse{}
-	response.Checksum = stubChecksum.Crc64Xor
-	response.TotalKvs = stubChecksum.TotalKvs
-	response.TotalBytes = stubChecksum.TotalBytes
-	return &response, nil
+func (c *TestChecksumClient) Checksum(ctx context.Context, startKey, endKey []byte, options ...rawkv.RawOption) (check rawkv.RawChecksum, err error) {
+	digest := crc64.New(crc64.MakeTable(crc64.ECMA))
+	checksum := rawkv.RawChecksum{}
+	allKeys := []string{}
+	for key := range c.store {
+		allKeys = append(allKeys, key)
+	}
+	sort.Strings(allKeys)
+	for _, key := range allKeys {
+		if bytes.Compare([]byte(key), startKey) < 0 {
+			continue
+		}
+		if len(endKey) > 0 && bytes.Compare([]byte(key), endKey) >= 0 {
+			continue
+		}
+		value := c.store[key]
+		digest.Reset()
+		digest.Write([]byte(key))
+		digest.Write([]byte(value))
+		checksum.Crc64Xor ^= digest.Sum64()
+		checksum.TotalKvs += 1
+		checksum.TotalBytes += (uint64)(len(key) + len(value))
+	}
+	return checksum, nil
+}
+func (c *TestChecksumClient) Scan(ctx context.Context, startKey, endKey []byte, limit int, options ...rawkv.RawOption) (keys [][]byte, values [][]byte, err error) {
+	retKeys := [][]byte{}
+	retValues := [][]byte{}
+	allKeys := []string{}
+	for key := range c.store {
+		allKeys = append(allKeys, key)
+	}
+	sort.Strings(allKeys)
+	for _, key := range allKeys {
+		if bytes.Compare([]byte(key), startKey) < 0 {
+			continue
+		}
+		if len(endKey) > 0 && bytes.Compare([]byte(key), endKey) >= 0 {
+			continue
+		}
+		value := c.store[key]
+		retKeys = append(retKeys, []byte(key))
+		retValues = append(retValues, []byte(value))
+	}
+	return retKeys, retValues, nil
 }
 
-func TestChecksumExecutor(t *testing.T) {
-	stubChecksum := getStubChecksum()
-	ctx := context.Background()
-	mockClient := &mockPdCLient{}
-	err := NewExecutor([]byte(""), []byte(""), kvrpcpb.APIVersion_V2, mockClient, 1).
-		Execute(ctx, stubChecksum, StorageChecksumCommand, nil)
-	require.Equal(t, err, nil)
-}*/
+func (c *TestChecksumClient) Close() error {
+	return nil
+}
+
+func GenerateTestData(keyIdx int64) (key, value string) {
```

Comment:
Nit: suggest to be `generateTestData` if it is used in this module only. Or move to a common module if it will be used by other test cases. The same to following functions.